### ISSUE TITLE
Test execution fails due to lazy containing multibyte characters

### ISSUE THREAD

**Issue by tukiyo320** [rspec-parameterized/rspec-parameterized-core#10]
# Problem
If the lazy contains multibyte characters such as Japanese, the test may fail.

# Sample code
https://github.com/tukiyo320/rspec-parameterized-core/commit/bd09082a8d2609e442da248c15b89e54910d2c33

# Situation
An additional condition occurs when one of the following two conditions is met

- A. multibyte characters are used in the description of the ancestor (including the parent) context
- B. There are other parameters in the parameter list to which lazy belongs that contain multibyte characters.

The error is as follows in both cases, but the location where the error occurs depends on the pattern.

```
Encoding::CompatibilityError: incompatible character encodings: UTF-8 and ASCII-8BIT
```

A. https://github.com/rspec/rspec-core/blob/f273314f575ab62092b2ad86addb6a3c93d6041f/lib/rspec/core/metadata.rb#L320
B. https://github.com/rspec-parameterized/rspec-parameterized-core/blob/94ed77992543f76d849805fd22f3fd38798b8f9d/lib/rspec/parameterized/core.rb#L114

# Cause

In both cases, the error seems to occur in the title generation part of the test case, and is probably caused by not reading the source code with the correct character encoding when interpreting the parameter list as a string.
For example, forcing the following encoding in `LazyArgs#inspect` works fine (of course, a deterministic encoding is not a good solution)

```
"#{@block.to_raw_source}".force_encoding('utf-8')
```

Tracing the cause, it seems that it is already treated as an unknown charset (ASCII-8BIT) inside the `proc_to_ast` gem that is used to read the source code.

https://github.com/joker1007/proc_to_ast/blob/60c95582828edb689d6f701543f8ae1b138139fc/lib/proc_to_ast.rb#L31

# Workaround
If you specify `case_name` in the parameter list, the problem processing will be skipped and it will work correctly.
Since there are not many tests that use lazy, this solution is not a problem, but we would like to see the hassle of specifying it and the difficulty in understanding the error eliminated.

# Remarks
- Sorry, but I haven't come up with a good solution.
- If you expect to specify `case_name`, it would be more helpful to put it in README or something.
- If you finally decide that the issue needs to be fixed with `proc_to_ast`, please comment on it (I will make an issue there).

**Comment by sue445** [comment]
@tukiyo320 Thanks for the detailed sample code. I understand your issue. 

I'm prefer to fix https://github.com/joker1007/proc_to_ast, but `rspec-parameterized-core` isn't bad either.

@joker1007 What do you think?

**Comment by joker1007** [comment]
I'm sorry for the late response.

I fixed the encoding mismatch in proc_to_ast gem.
And, I released v0.2.0.
https://github.com/joker1007/proc_to_ast/pull/4

Please use the version.

Thanks!

**Comment by sue445** [comment]
@joker1007 thank you!

@tukiyo320 Just to be safe, I have made sure to use proc_to_ast 0.2.0+ with #12 and published as v1.0.1 🚀  https://github.com/rspec-parameterized/rspec-parameterized-core/releases/tag/v1.0.1

**Comment by tukiyo320** [comment]
@sue445 @joker1007 Thank you all for your support!
